### PR TITLE
Convert native code to use N-API

### PR DESCRIPTION
**Pull request by marechal-p**
N-API claims to be ABI stable, meaning that a binary built using it
could be reused accross different Node.js versions.

Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

Fixes: #75

### CODE REVIEW

**Comment by simark** [comment]
I was looking over Paul's shoulder when he did this, so it LGTM.  It might be good (as another PR perhaps) to add a bit of comments to explain the less trivial bits, so newcomers understand why we do this and that.

**Comment by marechal-p** [comment]
Added comments, thank you again Simon.

File: src/native/pty.cc [hunk 1]
```diff
@@ -7,42 +7,46 @@
...
-static void create_pty(const v8::FunctionCallbackInfo<v8::Value> &args) {
-  v8::Isolate *isolate = args.GetIsolate();
+static Napi::Value create_pty(const Napi::CallbackInfo &info) {
+  Napi::Env env = info.Env();
 #ifdef WINDOWS
-  return _throw_exc(isolate, ".create_pty() is not supported on Windows");
+  // Windows does not supports TTYs.
+  _throw_exc(env, ".create_pty() is not supported on Windows");
 #else
+  // The fd will be closed on this scope exit.
```
Comment (simark) [inline src/native/pty.cc]: ... in the error cases.

**Comment by marechal-p** [comment]
Seems good to go, thanks.

**Comment by dschaefer** [comment]
Looks like tomorrow I'll finally have time to try this all out in vscode.

**Comment by marechal-p** [comment]
Left is to find a way to properly pack the extension for vscode, distributing the binaries for every platforms. Running VS Code from the `cdt-gdb-vscode repo` worked the last time I tried, but I don't think it can be distributed as is for everyone... At least the GDB console feature will fail on the other systems for a given build.

**Comment by marechal-p** [comment]
As a npm dependency it is fine though, as it will build for your platform on install.

**Pull request merged**